### PR TITLE
Bug fix/avoid cpp17 deprecations

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Fix various places where STL features have been used that have been deprecated in C++17 and removed in C++20.

- [x] :hammer: Refactoring/simplification

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: 3rdParty/immer/v0.6.2/tools/include/nonius.h++ [hunk 1]
```diff
@@ -659,7 +659,7 @@ namespace nonius {
             }
         };
         template <typename Sig>
-        using ResultOf = typename std::result_of<Sig>::type;
+        using ResultOf = typename std::invoke_result<Sig>::type;
```
Comment (maierlars) [inline 3rdParty/immer/v0.6.2/tools/include/nonius.h++:662]: Is that actually correct?

**Review by maierlars — APPROVED** [review]

File: 3rdParty/immer/v0.6.2/tools/include/nonius.h++ (hunk 1 above)
Comment (mpoeter) [reply 3rdParty/immer/v0.6.2/tools/include/nonius.h++:662]: Good catch - probably not! To make this correct we would have to change the template signature.

Comment (mpoeter) [reply 3rdParty/immer/v0.6.2/tools/include/nonius.h++]: Fixed.

**Review by markuspf — COMMENTED** [review]
Are the changes to `immer` unnecessary if #15072 is merged?

**Comment by maierlars** [comment]
Yes, immer is a c++11 library, so they can not use invoke_result.

**Pull request merged**